app/code/Magento/Inventory/Model/ResourceModel/SourceItem/DeleteMultiple.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model\ResourceModel\SourceItem;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Model\ResourceModel\SourceItem as SourceItemResourceModel;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+
+/**
+ * Implementation of SourceItem delete multiple operation for specific db layer
+ * Delete Multiple used here for performance efficient purposes over single delete operation
+ */
+class DeleteMultiple
```

Comment:
Hi @maghamed if we remove we will break the current delete function
behaviour. Because the current **Delete Multiple** removes all entries from the Database by only matching SKU.